### PR TITLE
refactor(contributor-guide): Updating the  contributor guidelines Readme

### LINKED ISSUE #1281: statement in developer guide is not very clear about the category of experiment
**What happened**:
The steps  mentioned in the developer guide is not very clear about the category of experiment 
Screenshot from 2020-03-09 17-00-50

**What you expected to happen**:
A  more detailed doc with a link to the directories and it should what to do if the subcategory is not there for an experiment.
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

**What this PR does / why we need it**:

- Updating the contributor guidelines Readme

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1281 

**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ksatchit** [comment]
@LaumiH @uditgaurav @atulabhi can you please share your thoughts on what else is missing / details to add after @ispeakc0de 's changes.to the devguide ? Thank you!

**Comment by uditgaurav** [comment]
I think we have covered almost all things, some suggestions on this are --  we can have one point about service Account permissions per experiment in the chart
Also some details about engine
Platform-based chaoslibs
--- 
In the end where we discussed how to create and test experiments. Also how to add it in chaoscharts and hub, we can have one more `how-to` saying how to add experiment in pipelines (raising PR in e2e repo) once we are done with the experiment changes.

We mention adding README with what way how it can be also taken to docs
cc: @ksatchit

File: contribute/developer_guide/README.md [hunk 1]
```diff
@@ -2,14 +2,14 @@
 
 The artefacts associated with a chaos-experiment are summarized below: 
 
-- Submitted in the litmuschaos/litmus repository, under the experiments/*chaos category* folder
+- Submitted in the litmuschaos/litmus repository, under the experiments/*chaos-category* folder i.e, `experiments/generic`
 
   - Experiment business logic (either as ansible playbook, or other). May involve creation of new or reuse of existing chaoslib
-  - Experiment Kubernetes job (executes the business logic, also called litmusbook)
+  - Experiment Kubernetes job (pass the experiment-specific chaos parameters, executes the business logic, also called litmusbook)
```
Comment (LaumiH) [inline contribute/developer_guide/README.md]: *passes
unclear if the job or the business logic is called litmusbook :D in the latter case please add it to above point

File: contribute/developer_guide/README.md [hunk 2]
```diff
@@ -2,14 +2,14 @@
 
 The artefacts associated with a chaos-experiment are summarized below: 
 
-- Submitted in the litmuschaos/litmus repository, under the experiments/*chaos category* folder
+- Submitted in the litmuschaos/litmus repository, under the experiments/*chaos-category* folder i.e, `experiments/generic`
 
   - Experiment business logic (either as ansible playbook, or other). May involve creation of new or reuse of existing chaoslib
```
Comment (LaumiH) [inline contribute/developer_guide/README.md]: Experiment business logic (either as Ansible Playbook, Golang, Python or Bash script, or other). May involve creating new or reusing an existing chaoslib (e.g. pumba, chaostoolkit).

File: contribute/developer_guide/README.md [hunk 3]
```diff
@@ -43,25 +43,47 @@ scaffolded files consist of placeholders which can then be filled as desired.
   $ cat attributes.yaml
```
Comment (LaumiH) [inline contribute/developer_guide/README.md]: the repository part should point to the litmuschaos/litmus repo. keep the example with the pod_delete to make it clear

Comment (LaumiH) [reply contribute/developer_guide/README.md]: I expect something like: To start automatically generating a new chart, edit the following file ... The above files will be generated automatically in the experiment folder.

File: contribute/developer_guide/README.md [hunk 4]
```diff
@@ -91,24 +121,20 @@ scaffolded files consist of placeholders which can then be filled as desired.
...
   -rw-r--r-- 1 ksatchit ksatchit   69 Oct 21 10:54 kill-hello-replicas-ansible-prerequisites.yml
   -rw-r--r-- 1 ksatchit ksatchit 2167 Oct 21 10:54 kill-hello-replicas-ansible-logic.yml
   
   ```
  
-  **Note**: In the `--generate_type` attribute, select the appropriate type of manifests to be generated, where, 
-  - `chart`: Just the chaos-chart metadata, i.e., chartserviceversion yaml 
-  - `experiment`: Chaos experiment artefacts belonging to a an existing OR new chart. 
-
 - Proceed with construction of business logic inside the `kill-hello-replicas-ansible-logic.yml` file, by making
```
Comment (LaumiH) [inline contribute/developer_guide/README.md]: rename file to pod delete

File: contribute/developer_guide/README.md (hunk 1 above)
Comment (LaumiH) [reply contribute/developer_guide/README.md]: also, not sure if litmusbook is still an adequate name for it, as I understand this came from ansible playbook, but not only ansible is used now

**Pull request merged**